rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -411,20 +411,39 @@ impl Controller {
         Ok(())
     }
 
-    fn reload_resource_list(&mut self) -> Result<()> {
-        let Some(session) = &self.session else {
-            tracing::warn!("got notified to update resources but there is no session");
-            return Ok(());
-        };
-        let resources = session.callback_handler.resources.load();
-        let actor_name = &self
-            .auth
-            .session()
-            .ok_or_else(|| anyhow!("connlib is signed in but auth module isn't"))?
-            .actor_name;
-        self.app
-            .tray_handle()
-            .set_menu(system_tray_menu::signed_in(actor_name, &resources))?;
+    /// Returns a new system tray menu
+    fn build_system_tray_menu(&self) -> Result<tauri::SystemTrayMenu> {
+        Ok(if let Some(auth_session) = self.auth.session() {
+            if let Some(connlib_session) = &self.session {
+                // TODO: Modify CallbackHandler so it can tell us whether the tunnel is really ready.
+                let tunnel_ready = true;
+                if tunnel_ready {
```

Comment:
we can leave the todo but I'd rather remove this `if true`

Reply:
I wanted to make space for it in the control flow. Luckily the PR for this feature landed already so I just implemented that, too, so it'll say "Signing in..." after auth is done and before connlib gives us `on_tunnel_ready`